File: 2021/09/19/SetMismatch.java
/**
 * https://leetcode.com/problems/set-mismatch/
 * @author Andrew Jarombek
 * @since 9/19/2021
 */

import java.util.HashSet;
import java.util.Set;

public class SetMismatch {
    public int[] findErrorNums(int[] nums) {
        int[] result = new int[2];
        Set<Integer> numSet = new HashSet<>();

        for (int i = 0; i < nums.length; i++) {
            if (numSet.contains(nums[i])) {
                result[0] = nums[i];
            }

            numSet.add(nums[i]);
        }

        for (int i = 1; i <= nums.length; i++) {
            if (!numSet.contains(i)) {
                result[1] = i;
                break;
            }
        }

        return result;
    }
}

File: 2022/04/06/setMismatch.ts
/**
 * https://leetcode.com/problems/set-mismatch/
 * @author Andrew Jarombek
 * @since 4/6/2022
 */

function findErrorNums(nums: number[]): number[] {
    nums.push(nums.length + 1);
    nums.sort((a, b) => a - b);
    let prev = 0;
    let result = [0, 0];
    let i = 0;

    while (result[0] == 0 || result[1] == 0) {
        let current = nums[i];

        if (current == prev + 2) {
            result[1] = prev + 1;
        } else if (current == prev) {
            result[0] = current;
        }

        prev = current;
        i++;
    }

    return result;
}